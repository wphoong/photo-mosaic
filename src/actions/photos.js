import uuid from "uuid";

// ADD PHOTO

export const addPhoto = (
  {
    photoLink = "",
    title = "",
    createdAt = 0,
    description = ""
  } = {}) => ({
    type: "ADD_PHOTO",
    photo: {
      id: uuid(),
      photoLink,
      title,
      createdAt,
      description
    }
  }
);

export const editPhoto = (id, updates) => ({
  type: "EDIT_PHOTO",
  id,
  updates
});

export const removePhoto = ({id} = {}) => ({
  type: "REMOVE_PHOTO",
  id
});

// export const startAddPhoto = (photoData = {}) => {
//   return (dispatch, getState) => {
//     const uid = getState().auth.uid;
//     const {
//       photoLink = "",
//       title = "",
//       description = "",
//       createdAt = 0
//     } = photoData;
//     const photo = { photoLink, title, description, createdAt };
    
//     return dispatch(addPhoto({
//       id: ref.key,
//       ...photo 
//     }));
//   };
// };