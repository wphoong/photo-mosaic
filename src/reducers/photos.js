const photosReducerDefaultState = [];

const photosReducer = (state = photosReducerDefaultState, action) => {
  switch (action.type) {
    case "ADD_PHOTO":
      return [
        ...state,
        action.photo
      ];
    default:
      return state;
  }
};

export default photosReducer;