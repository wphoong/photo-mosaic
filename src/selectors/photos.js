import moment from "moment";

const getVisiblePhotos = (photos, { text, sortBy, startDate, endDate}) => {
  return photos.filter((photo) => {
    const createdAtMoment = moment(photos.createdAt);
    const setTextMatch = photo.title.toLowerCase().includes(text.toLowerCase());
    const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, "day") : true;
    const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, "day") : true;

    return setTextMatch && startDateMatch && endDateMatch;
  }).sort((a, b) => {
      if (sortBy === "descend") {
        return b.createdAt - a.createdAt;
      } else {
        return a.createdAt - b.createdAt;
      }
    }
  );
};

export default getVisiblePhotos;
