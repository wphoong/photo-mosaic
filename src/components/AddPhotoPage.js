import React from "react";
import { connect } from "react-redux";
import PhotoForm from "./PhotoForm.js";
import { startAddPhoto } from "../actions/photos.js";

export class AddPhotoPage extends React.Component {
  onSubmit = (photo) => {
    this.props.startAddPhoto(photo);
    this.props.history.push("/");
  }
  render() {
    return (
      <div className="content-container">
        <div>
          <h1>Add Photo</h1>
        </div>
        <div>
          <PhotoForm onSubmit={this.onSubmit} />
        </div>
      </div>
    );
  }
}

const mapStateToDispatch = (dispatch) => ({
  startAddPhoto: (photo) => dispatch(startAddPhoto(photo))
});

export default connect(undefined, mapStateToDispatch)(AddPhotoPage);