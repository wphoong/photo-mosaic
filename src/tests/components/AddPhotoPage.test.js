import React from "react";
import { shallow } from "enzyme";
import { AddPhotoPage } from "../../components/AddPhotoPage.js";
import photos from "../fixtures/photos.js";

let wrapper, addPhoto, history;

beforeEach(() => {
  addPhoto = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(<AddPhotoPage addPhoto={addPhoto} history={history} />);
});

test("it should render the add photo page", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should handle on submit", () => {
  wrapper.find("PhotoForm").prop("onSubmit")(photos[0]);
  expect(addPhoto).toHaveBeenLastCalledWith(photos[0]);
  expect(history.push).toHaveBeenLastCalledWith("/");
});