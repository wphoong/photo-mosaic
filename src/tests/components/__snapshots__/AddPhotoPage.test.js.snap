// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render the add photo page 1`] = `
<div>
  <div>
    <h1>
      Add Photos
    </h1>
  </div>
  <div>
    <PhotoForm
      onSubmit={[Function]}
    />
  </div>
</div>
`;
